internal/controller/execplan.go

```diff
@@ -131,6 +131,17 @@ func (r *TemporalWorkerDeploymentReconciler) executePlan(ctx context.Context, l
 			}); err != nil { // would be cool to do this atomically with the update
 				return fmt.Errorf("unable to update metadata after setting ramping deployment: %w", err)
 			}
+		} else if ramp := vcfg.rampPercentage; ramp == 0 {
```

Comment:
I don't think you actually need this else if ramp == 0 case, because if `ramp==0`, the `SetRampingVersion` call above would handle things the same way (if you also remove the `ramp>0` condition from the above block). I think we still want to update the metadata even when setting ramp to 0, so it would be desirable to put all that in one block for that reason as well.

Reply:
If the structure was
```
if vcfg.SetCurrent {
  SetCurrent
  UpdateMetadata
} else {
  SetRamping
  UpdateMetadata
}
```
I think that works?

Reply:
or even
```
if vcfg.SetCurrent {
  SetCurrent
} else {
  SetRamping
}
UpdateMetadata
```

Reply:
Sidenote that i can't comment on because you didnt change that line. Now that there is a list of vcfgs, do you still want to `return` if there is an error setting ramp? or setting current? Or do we want to log the error and continue to the next vcfg in the list. It's too late in my work day for me to fully think this through right now, but something we should think about!